### PR TITLE
First pass on querying controls in basic api

### PR DESCRIPTION
**Pull request by jimon**
Still think we should try method overloads from https://github.com/Unity-Technologies/InputSystem/issues/1559#issuecomment-1248152830

### CODE REVIEW

File: Packages/com.unity.inputsystem/InputSystem/API/Input.cs [hunk 1]
```diff
@@ -254,48 +258,224 @@ public static partial class Input
         /// </summary>
         public static IReadOnlyList<Gamepad> gamepads { get; }
 
+#endif
+
+        private static ButtonControl GetButtonControl(Inputs input)
+        {
+            // WHAT THE FCK!? or .. can we do better?
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/InputSystem/API/Input.cs]: A minor improvement (for enums in the keyboard range at least) might be to move the key mapping out of here and into a dedicated function. Eg.
`
if (input < Inputs.Key_MAX) {
    return Keyboard.current != null ? Keyboard.current[KEYMAP_FUN(input)] : null;
}
`

File: Packages/com.unity.inputsystem/InputSystem/API/Input.cs [hunk 2]
```diff
@@ -254,48 +258,224 @@ public static partial class Input
...
         /// </summary>
         /// <param name="input"></param>
         /// <returns></returns>
         /// <remarks>
-        /// This will look at all devices of the appropriate type (which will depend on the specified Inputs)
-        /// and return true if the control was released in the current frame on any of them.
+        ///     This will look at all devices of the appropriate type (which will depend on the specified Inputs)
+        ///     and return true if the control was released in the current frame on any of them.
         /// </remarks>
         public static bool IsControlUp(Inputs input)
```
Comment (jamesmcgill) [inline Packages/com.unity.inputsystem/InputSystem/API/Input.cs:707]: Still got confused by the naming here (but maybe just me) as it doesn't imply it's temporal in nature.
I.e. this is going to return false the majority of the time, even when keys are not being pressed (and are 'up').

**Review by jamesmcgill — APPROVED** [review]
nice!

File: Packages/com.unity.inputsystem/InputSystem/API/Input.cs (hunk 1 above)
Comment (ekcoh) [reply Packages/com.unity.inputsystem/InputSystem/API/Input.cs]: Agree with single null check and early return on that.

But why arent we just setting the enum values to the enum value of the underlying mapped type? If you want a translation table for some reason thats fine, but maybe generate it in that case? Or put it in an extension method for the enum and return just value (and then translations can be added there on a need basis)?

**Review by andrew-oc — CHANGES_REQUESTED** [review]
Xmldoc comments should be updated for public methods as part of this PR

**Pull request merged**